src/sentry/incidents/typings/metric_detector.py

```diff
@@ -162,3 +163,21 @@ def from_legacy_models(
             new_status=new_status,
             metric_value=metric_value,
         )
+
+
+@dataclass
+class OpenPeriodContext:
+    """
+    We want to eventually delete this class. it serves as a way to pass data around
+    that we used to use `incident` for.
+    """
+
+    date_started: datetime
+    date_closed: datetime | None
+
```

Comment:
General note, but if you find repeated actions/operations in the usages of this data that could be extracted into methods on this dataclass, it might be worth doing.